packages/components/src/input-control/reducer/reducer.ts

```diff
@@ -109,7 +109,11 @@ function inputControlStateReducer(
 		}
 
 		if ( action.payload.event ) {
-			nextState._event = action.payload.event;
+			const isDrag =
+				action.type === actions.DRAG_START ||
+				action.type === actions.DRAG ||
+				action.type === actions.DRAG_END;
+			nextState._event = isDrag ? action.payload : action.payload.event;
```

Comment:
This is the core difference. When the action one of the drag ones, the full gesture state becomes `_event` on the new state. It looks enough like an event I think and comes with a type we can use 🤷.